data/slash/multipleHeaderSigningProof.go

```diff
@@ -75,36 +96,75 @@ func NewMultipleSigningProof(slashResult map[string]SlashingResult) (MultipleSig
 		return nil, data.ErrNilSlashResult
 	}
 
-	pubKeys := make([][]byte, 0, len(slashResult))
-	levels := make(map[string]ThreatLevel, len(slashResult))
-	headers := make(map[string]HeadersV2, len(slashResult))
+	headersInfo, err := getAllUniqueHeaders(slashResult)
+	if err != nil {
+		return nil, err
+	}
+	sortedHeaders, err := sortAndGetHeadersV2(headersInfo)
```

Comment:
could it be renamed to getSortedHeadersV2?

Reply:
Yes, renamed it